estimagic/inference/shared.py

```diff
@@ -43,20 +47,21 @@ def transform_covariance(
     processed_constraints, processed_params = process_constraints(constraints, params)
     free_index = processed_params.query("_internal_free").index
 
+    if isinstance(internal_cov, pd.DataFrame):
```

Comment:
Similar here. I would be wondering whether it is nicer to handle these issues outside?

Reply:
In this case it would be very hard.